cast/src/starknet_commands/declare.rs

```diff
@@ -14,41 +13,110 @@ use starknet::{
     signers::LocalWallet,
 };
 use std::sync::Arc;
+use std::process::Command;
+use serde::Deserialize;
+use std::path::PathBuf;
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetArtifacts {
+    version: u32,
+    contracts: Vec<ScarbStarknetContract>,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetContract {
+    id: String,
+    package_name: String,
+    contract_name: String,
+    artifacts: ScarbStarknetContractArtifact,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetContractArtifact {
+    sierra: PathBuf,
+    casm: Option<PathBuf>,
+}
 
 #[derive(Args)]
 #[command(about = "Declare a contract to starknet", long_about = None)]
 pub struct Declare {
-    /// Path to the sierra compiled contract
-    #[clap(short = 's', long = "sierra-contract-path")]
-    pub sierra_contract_path: Utf8PathBuf,
-
-    /// Path to the casm compiled contract
-    #[clap(short = 'c', long = "casm-contract-path")]
-    pub casm_contract_path: Utf8PathBuf,
+    /// contract name
+    pub contract: String,
 
     /// Max fee for the transaction. If not provided, max fee will be automatically estimated
     #[clap(short, long)]
     pub max_fee: Option<u128>,
 }
 
 pub async fn declare(
-    sierra_contract_path: &Utf8PathBuf,
-    casm_contract_path: &Utf8PathBuf,
+    contract_name: &str,
     max_fee: Option<u128>,
     account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
 ) -> Result<DeclareTransactionResult> {
+    let contract_name: String = contract_name.to_string();
+    let command_result = Command::new("scarb")
+        .current_dir(std::env::current_dir().context("Failed to obtain current dir")?)
+        .arg("build")
+        .output()
+        .context("Failed to start building contracts with Scarb")?;
+    let result_code = command_result.status.code().context("failed to obtain status code from scarb build")?;
+    if result_code != 0 {
+        panic!("scarb build returned non-zero exit code: {}", result_code);
```

Comment:
Consider returning an error (`bail`) instead of `panic` for non-invariants.

Reply:
Is Scarb output passed to the UI? Especially in case of failure?
Consider 
```
.stdout(Stdio::piped()).stderr(Stdio::piped())
```

Reply:
[address feedback](https://github.com/software-mansion/protostar/pull/2138/commits/1fbe96f5d95497671ce5af742d0fc2b906d5f63d)